client/src/contexts/AuthContext.js

```diff
@@ -34,30 +33,57 @@ const Provider = ({ children }) => {
   let history = useHistory();
   let location = useLocation();
 
-  const signUp = async ({ email, password }) => {
-    // TODO call back end
-    dispatch({ type: "signUp", payload: { user: fakeUser } });
-    history.push("/");
+  const signUp = async (user) => {
+    try {
+      const { data } = await API.post("/users", user);
+      dispatch({ type: "signUp", payload: { user: data.user } });
+      history.push("/");
+    } catch (error) {
+      alert(error.response.data.message);
+    }
   };
   const login = async ({ email, password }) => {
-    // TODO call back end
-    dispatch({ type: "login", payload: { user: fakeUser } });
-    // We are trying to redirect the user to the page he was attempting to visit before he got redirected to login
-    let { from } = location.state || { from: { pathname: "/" } };
-    history.replace(from);
+    try {
+      const { data } = await API.post("/users/login", { email, password });
+      dispatch({ type: "login", payload: { user: data.user } });
+      let { from } = location.state || { from: { pathname: "/" } };
+      history.replace(from);
+    } catch (error) {
+      alert(error.response.data.message);
+    }
   };
 
   const clearErrorMessage = () => dispatch({ type: "clearErrorMessage" });
 
   const signOut = async () => {
-    // TODO call back end
-    dispatch({ type: "signOut" });
-    history.push("/login");
+    try {
+      await API.post("/users/logout");
+      dispatch({ type: "signOut" });
+      history.push("/login");
+    } catch (error) {
+      alert(error.response.data.message);
+    }
+  };
+
+  const refreshLoggedInUser = async () => {
+    try {
+      const { data } = await API.get("/users/login");
```

Comment:
Should this actually be refresh token route?

Reply:
I see your point, will consolidate refreshing the user object and getting a new access token to one route, let me know if this is acceptable or not